RailwayBooking.java

```diff
@@ -3,36 +3,40 @@
 public class RailwayBooking {
     private DataBase db = new DataBase();
     public List<Passenger> tempPassengers = new ArrayList<>();
+    public List<Passenger> waitingListPassengers = new ArrayList<>();
     public HashMap<String, Integer> bookedTickets = new HashMap<>();
     public List<User> tempUsers = new ArrayList<>();
     int seatNumber = 1;
     int count = 1;
 
     public boolean bookTheTicket(String username, String name, String gender, int age, BerthPreference berthPreference, ClassType classType, String from, String to, String trainName) {
         List<Train> availableTrains = checkTrains(from, to);
+        boolean isFound = false;
         for (Train t : availableTrains) {
-            if (trainName.equals(t.getTrainName())) {
-                if (isSeatAvailable(t, classType, berthPreference)) {
-                    bookSeat(t, classType, berthPreference);
-                    Passenger newPassenger = new Passenger(name, gender, age, berthPreference, classType);
-                    db.passengers.add(newPassenger);
-                    String ticketKey = generateTicketKey(to, classType, seatNumber++, berthPreference);
-                    System.out.println("Ticket id: " + ticketKey);
-                    bookedTickets.put(ticketKey, count++);
-
-                    // Add ticket to user
-                    for (User user : db.getUsers()) {
-                        if (user.getUserName().equals(username)) {
-                            user.addTicket(ticketKey);
-                        }
-                    }
+            if (trainName.equals(t.getTrainName()) && isSeatAvailable(t, classType, berthPreference)) {
+            
+                bookSeat(t, classType, berthPreference);
+                Passenger newPassenger = new Passenger(name, gender, age, berthPreference, classType);
+                db.passengers.add(newPassenger);
+                String ticketKey = generateTicketKey(to, classType, seatNumber++, berthPreference);
+                System.out.println("Ticket id: " + ticketKey);
+                bookedTickets.put(ticketKey, count++);
 
-                    return true;
-                } 
+                for (User user : db.getUsers()) {
+                    if (user.getUserName().equals(username)) {
+                        user.addTicket(ticketKey);
+                    }
+                }
+                isFound = true;
```

Comment:
what is the usecase?

Reply:
The "isFound" variable is used to store confirmed tickets, and waiting list tickets are stored in a separate list.

Reply:
change the passenger class to ticket class and have the ticket status property.